### PR TITLE
Feat/1463

### PR DESCRIPTION
**Pull request by Kolezhanchik**

### CODE REVIEW

File: app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import TableAWSRoles from '@/components/table/TableAWSRoles';
+import {
+  getSubGroupMembersByLicencePlateAndName,
+  getPublicCloudProjectUsers,
+  User,
+} from '@/app/api/public-cloud/aws-roles/routes';
+import TableBodyAWSRoles from '@/components/table/TableBodyAWSRoles';
+export default async function ProductAWSRoles() {
+  const users = await getSubGroupMembersByLicencePlateAndName('eu9cfk', 'Admins', 'Admin');
```
Comment (junminahn) [inline app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx:9]: are the arguments sample data?

File: app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const registryUsers = await getPublicCloudProjectUsers('A1VEcjg');
```
Comment (junminahn) [inline app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx:10]: are the arguments sample data?

File: app/api/public-cloud/aws-roles/routes.ts [hunk 3]
```diff
@@ -144,7 +172,7 @@ const findObjectByValueSubstring = (array: Group[], key: keyof Group, value: any
   return array.filter((obj) => obj[key].includes(value));
 };
 
-export async function getProductAWSRoles(licencePlate: string = 'eu9cfk'): Promise<Group[] | undefined> {
+export async function getProductAWSRoles(licencePlate: string): Promise<Group[] | undefined> {
   const keyClockGroups = await getGroups;
```
Comment (junminahn) [inline app/api/public-cloud/aws-roles/routes.ts:176]: The usage seems little bit interesting... maybe we can convert `getGroups` as an actual function?

File: app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import TableAWSRoles from '@/components/table/TableAWSRoles';
+import {
+  getSubGroupMembersByLicencePlateAndName,
+  getPublicCloudProjectUsers,
+  User,
+} from '@/app/api/public-cloud/aws-roles/routes';
+import TableBodyAWSRoles from '@/components/table/TableBodyAWSRoles';
+export default async function ProductAWSRoles() {
```
Comment (junminahn) [inline app/(request)/public-cloud/edit/[licencePlate]/roles/Admins/page.tsx:8]: I believe this page is only for the `admin` user?
Let's add the session & permission checks.

**Review by junminahn — COMMENTED** [review]
I added a few comments for now.
I will pull the changes on my local environment to test it more.
Thanks!

File: app/api/public-cloud/aws-roles/routes.ts (hunk 3 above)
Comment (Kolezhanchik) [reply app/api/public-cloud/aws-roles/routes.ts:176]: could you, please, provide more explanations with that, thank you

Comment (junminahn) [reply app/api/public-cloud/aws-roles/routes.ts:176]: yes, can you convert `getGroups` as a `Function`, it is a `Promise` type now.

File: components/tabs/PublicUsersTabs.tsx [hunk 5]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
-            </div>
-          </div>
-        </div>
-      </div> */}
+    <div>
+      <span className="isolate inline-flex rounded-md shadow-sm py-2 px-4">
+        {tabs.map((tab, index) => (
+          <Link
+            key={index}
+            href={`${pathname.split('/').splice(0, 5).join('/')}/${tab.href}`}
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:38]: Can we make a helper function to increase the readability?
```js
function extractPathSegments(pathname, count) {
    return pathname.split('/').splice(0, count).join('/');
}
```

File: components/tabs/PublicUsersTabs.tsx [hunk 6]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
-    name: 'ADMINS',
-    href: 'admins',
+    name: 'Admins',
+    href: 'Admins',
   },
   {
-    name: 'BILLING VIEWERS',
-    href: 'billing-viewers',
+    name: 'Billing Viewers',
+    href: '/BillingViewers',
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:14]: It seems like `/` is an extra.

File: components/tabs/PublicUsersTabs.tsx [hunk 7]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
   },
   {
-    name: 'DEVELOPERS',
-    href: 'developers',
+    name: 'Developers',
+    href: '/Developers',
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:18]: It seems like `/` is an extra.

File: components/tabs/PublicUsersTabs.tsx [hunk 8]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
   },
   {
-    name: 'SECURITY AUDITORS',
-    href: 'security-auditors',
+    name: 'Security Auditors',
+    href: '/SecurityAuditors',
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:22]: It seems like `/` is an extra.

File: components/tabs/PublicUsersTabs.tsx [hunk 9]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
   },
   {
-    name: 'VIEWERS',
-    href: 'viewers',
+    name: 'Viewers',
+    href: '/Viewers',
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:26]: It seems like `/` is an extra.

File: components/tabs/PublicProductTabs.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          ))}
+        </select>
+      </div>
+      <div className="hidden md:block  justify-start">
+        <div className="border-b border-gray-200">
+          <div className="w-full -mb-px flex justify-between items-center" aria-label="Tabs">
+            <div className=" -mb-px flex justify-start">
+              <Link
+                key={'PUBLIC CLOUD EDIT'}
+                href={`${pathname.split('/').splice(0, 4).join('/')}/product`}
```
Comment (junminahn) [inline components/tabs/PublicProductTabs.tsx:46]: Can we make a helper function to increase the readability?
```js
function extractPathSegments(pathname, count) {
    return pathname.split('/').splice(0, count).join('/');
}
```

File: components/tabs/PublicProductTabs.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                    ? "relative border-bcorange text-bcblue before:content-[''] before:absolute before:w-2/4 before:border-b-3 before:border-bcorange before:bottom-0 before:left-1/2 before:-translate-x-1/2"
+                    : "relative border-transparent text-gray-300 hover:before:content-[''] hover:before:absolute hover:before:w-2/4 hover:before:border-b-3 hover:before:border-gray-300 hover:before:bottom-0 hover:before:left-1/2 hover:before:-translate-x-1/2",
+                  'lg:ml-20 w-50 py-5 text-center font-bcsans text-lg font-bold',
+                )}
+              >
+                PUBLIC CLOUD USER EDIT
+              </Link>
+              <Link
+                key="PUBLIC CLOUD USER ROLES"
+                href={`${pathname.split('/').splice(0, 4).join('/')}/roles/Admins`}
```
Comment (junminahn) [inline components/tabs/PublicProductTabs.tsx:58]: Can we make a helper function to increase the readability?
```js
function extractPathSegments(pathname, count) {
    return pathname.split('/').splice(0, count).join('/');
}
```

File: components/tabs/PublicUsersTabs.tsx [hunk 12]
```diff
@@ -6,69 +6,48 @@ import { usePathname } from 'next/navigation';
...
-      </div> */}
+    <div>
+      <span className="isolate inline-flex rounded-md shadow-sm py-2 px-4">
+        {tabs.map((tab, index) => (
+          <Link
+            key={index}
+            href={`${pathname.split('/').splice(0, 5).join('/')}/${tab.href}`}
+            type="button"
+            className={classNames(
+              (pathname.split('/').includes(tab.href) ? 'bg-gray-200 hover:none' : 'bg-white hover:bg-gray-100') +
```
Comment (junminahn) [inline components/tabs/PublicUsersTabs.tsx:41]: I am concerns of its reliability of the check; let's explicitly check the position of the segments as well.
We could also make it as a helper function.
```js
function hasSegmentAtPosition(pathname, targetSegment, position) {
    const pathSegments = pathname.split('/');
    
    // Check if the specified position is within the array bounds
    if (position >= 0 && position < pathSegments.length) {
        return pathSegments[position] === targetSegment;
    }
    
    // If the position is out of bounds, return false
    return false;
}
```

**Review by junminahn — COMMENTED** [review]
I went through a second round.
In general, let's follow the `route naming convention` we have, which is `kebab-case`.
https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case

**Review by junminahn — APPROVED** [review]

**Pull request merged**